libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.m

```diff
@@ -0,0 +1,301 @@
+    BOOL result = NO;
+    if (![self isReservedName:windowName]) {
+        [self.namedWindows removeObjectForKey:windowName];
+        result = YES;
+    }
+    return result;
+}
+
+- (SFSDKWindowContainer *)windowWithName:(NSString *)name {
+    return [self.namedWindows objectForKey:name];
+}
+
+- (void)addDelegate:(id<SFSDKWindowManagerDelegate>)delegate
+{
+    @synchronized (self) {
+        [_delegates addObject:[NSValue valueWithNonretainedObject:delegate]];
+    }
+}
+
+- (void)removeDelegate:(id<SFSDKWindowManagerDelegate>)delegate
+{
+    @synchronized (self) {
+        [_delegates removeObject:[NSValue valueWithNonretainedObject:delegate]];
+    }
+}
+
+- (void)enumerateDelegates:(void (^)(id<SFSDKWindowManagerDelegate> delegate))block
+{
+    @synchronized(self) {
+        [_delegates.allObjects enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+            id<SFSDKWindowManagerDelegate> delegate = [obj nonretainedObjectValue];
+            if (delegate) {
+                if (block) block(delegate);
+            }
+        }];
+    }
+}
+#pragma mark - SFSDKWindowContainerDelegate
+- (void)windowEnable:(SFSDKWindowContainer *_Nonnull)window animated:(BOOL)animated withCompletion:(void (^)(void))completion{
+
+    [self enumerateDelegates:^(id<SFSDKWindowManagerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowManager:willEnableWindow:)]){
+            [delegate windowManager:self willEnableWindow:window];
+        }
+    }];
+    
+    if (animated) {
+        __weak typeof (self) weakSelf = self;
+        UIViewPropertyAnimator *animator = [[UIViewPropertyAnimator alloc] initWithDuration:0.25 curve:UIViewAnimationCurveEaseInOut animations:^{
+            [weakSelf makeOpaqueWithCompletion:window completion:completion];
+        }];
+        [animator startAnimation];
+    } else {
+        [self makeOpaqueWithCompletion:window completion:completion];
+    }
+}
+
+- (void)windowDisable:(SFSDKWindowContainer *)window animated:(BOOL)animated withCompletion:(void (^)(void))completion{
+    [self enumerateDelegates:^(id<SFSDKWindowManagerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowManager:willDisableWindow:)]){
+            [delegate windowManager:self willDisableWindow:window];
+        }
+    }];
+   
+    if (animated) {
+        __weak typeof (self) weakSelf = self;
+        UIViewPropertyAnimator *animator = [[UIViewPropertyAnimator alloc] initWithDuration:0.25 curve:UIViewAnimationCurveEaseInOut animations:^{
+            [weakSelf makeTransparentWithCompletion:window completion:completion];
+        }];
+        
+        [animator startAnimation];
+    } else {
+        [self makeTransparentWithCompletion:window completion:completion];
+    }
+    
+}
+
+
+#pragma mark - private methods
+- (void)makeTransparentWithCompletion:(SFSDKWindowContainer *)window completion:(void (^)(void))completion {
+    window.window.alpha = 0.0; //make Transparent
```

Comment:
@wmathurin based on the previous explanation, the logic should be applied to only sdk windows. For us the requirement is clear, i.e. visible or not. As much as this code looks generic the alpha attributes are only applied to snapshot, passcode  and auth windows.